spirv_glsl.cpp

```diff
@@ -14865,11 +14865,19 @@ void CompilerGLSL::emit_block_chain(SPIRBlock &block)
 
 		size_t num_blocks = block_declaration_order.size();
 
-		const auto to_case_label = [](uint32_t literal, bool is_unsigned_case) -> string {
-			return is_unsigned_case ? convert_to_string(literal) : convert_to_string(int32_t(literal));
+		// Precompute the is_32bit_int since it's block based, not per-case
+		// based so we then move the bool to the lambda.
+		const bool is_32bit_int = (type.basetype == SPIRType::Int);
```

Comment:
This misses the case where the switch condition is 16-bit.

Reply:
Maybe I should open other issue for this but, [the following spirv is valid](https://gist.github.com/etra0/0900c18651dfa61092a77e8f92d0ad3d#file-switch-spvasm), which spirv-cross will generate the frag just fine, but then it won't compile back since `glslc` requires explicitly that switch conditions are int, and in the spec I think it's specified that it should be `int` or `uint`